learning_phase/go-chat/client/client.go

```diff
@@ -83,34 +119,98 @@ func (cli *client) listenForServerMessages(ctx context.Context, conn net.Conn, m
 	msg_channel (in the latter case msg_channel wouldn't be required). Should also be able to receive termination signals
 	from getServerMessage and cancellation of context from Run(), and terminate accordingly.
 	*/
-
+	rcvd_msg := make(chan string)
+	exit_chan := make(chan bool)
+	terminate_chan := make(chan bool)
+	exit_chan <- false
+	//check := <-exit_chan
+	//var message string
+	for {
+		go getServerMessage(conn, rcvd_msg, exit_chan, terminate_chan)
+		message := <-rcvd_msg
+		fmt.Println("\n", message)
+		select {
+		case <-ctx.Done():
+			fmt.Println("Client-Side Termination")
+			final_term_chan <- true
+			return
+
+		case <-terminate_chan:
+			{
+				final_term_chan <- true
+				fmt.Println("Server-Side Termination")
+				return
+			}
+
+		}
+
+	}
 
 }
 
-
-func getClientMessage(sc bufio.Scanner, conn net.Conn, rcvd_msg chan string) {
+func getClientMessage(sc *bufio.Scanner, rcvd_msg chan string) {
 
 	/**
 	Called by listenForClientMessages defined below. Should listen for messages entered by the user and
 	write it to the rcvd_msg channel
 	*/
+	var finalmessage string
+	for sc.Scan() {
+		message := sc.Text()
+		finalmessage += message
+	}
+	rcvd_msg <- finalmessage
 
 }
 
-func (cli *client) listenForClientMessages(ctx context.Context, sc bufio.Scanner, conn net.Conn, final_term_chan chan bool) {
+func (cli *client) listenForClientMessages(ctx context.Context, sc *bufio.Scanner, conn net.Conn, final_term_chan chan bool) {
 
 	/**
 	Called by Run(). Uses getClientMessage to read messages entered by the user. When a message has been read, it should
 	be parsed and be made to follow the protocol format, and sent to the server. Should be able to handle cancellation of context
 	from Run() too.
 	*/
+	rcvd_msg := make(chan string)
+	for {
+		go getClientMessage(sc, rcvd_msg)
+		//var structure message_struct
+		var sendmessage string
+		select {
+		case <-ctx.Done():
+			final_term_chan <- true
+			return
+		case message := <-rcvd_msg:
+			splitmessage := strings.Split(message, "~")
+			mtype := splitmessage[0]
+			switch mtype {
+			case "Authenticate":
+				sendmessage = string("0~" + cli.Username + "~" + splitmessage[1])
+			case "Unicast":
+				//structure = message_struct{0, cli.Username, splitmessage[1], splitmessage[2]}
+				sendmessage = string("1~" + cli.Username + "~" + splitmessage[1] + "~" + splitmessage[2])
+			case "Broadcast":
+				//structure = message_struct{1, cli.Username, "N/A", splitmessage[1]}
+				sendmessage = string("2~" + cli.Username + "~N/A~" + splitmessage[1])
+			case "Terminate":
+				//structure = message_struct{2, cli.Username, "N/A", "N/A"}
+				sendmessage = string("3~" + cli.Username + "~N/A~N/A")
+			}
+
+		}
+		//feedbuffer := new(bytes.Buffer)
+		//gobobj := gob.NewEncoder(feedbuffer)
+		//gobobj.Encode(structure)
+		//conn.Write(feedbuffer.Bytes())
+		conn.Write([]byte(sendmessage + "\n"))
```

Comment:
I think the sendmessage needs to be padded to reach 256 bytes, if you're specifying that all messages to be fixed at that size